google/cloud/firestore_v1/base_batch.py

```diff
@@ -33,14 +33,22 @@ class BaseWriteBatch(object):
     Args:
         client (:class:`~google.cloud.firestore_v1.client.Client`):
             The client that created this batch.
+        write_ctx (bool):
+            Controls whether this instance should call `commit` or `write` upon
+            exiting as a context manager. This parameter has no impact if you
+            do not use the instance as a context manager.
     """
 
-    def __init__(self, client) -> None:
+    def __init__(self, client, write_ctx: bool = False) -> None:
         self._client = client
+        self._write_ctx = write_ctx
         self._write_pbs = []
         self.write_results = None
         self.commit_time = None
 
+    def __len__(self):
+        return len(self._write_pbs)
+
```

Comment:
How do you imagine `len(batch)` being used?

Reply:
`BulkWriter` looks to cut off batch sizes at 20 operations, so this allows that code to evaluate `len(batch)` instead of `len(batch._write_pbs)`.